Basic gem setup, Client, and Configuration

edwardloveall opened the pull request:
We now have a basic gem skeleton, tests, a Client class and a Configuration class.

Client takes a string like `'hockey/nhl/game_logs'` and calls the Stattleship api with correct headers and security:

``` ruby
Stattleship::Client.new(path: 'hockey/nhl/game_logs')

#> JSON Response...
```

Configuration can be used like most configuration patterns:

``` ruby
Stattleship.configure do |config|
  config.api_token = 'abc123'
end
```

Currently it can take two values `api_token` and `base_uri`. `base_uri` is never meant to be set by the user, but it gives us a choke point to access and set that url if we need it.

edwardloveall committed:
Initial gem skeleton

edwardloveall committed:
Client

The client is responsible for connecting to stattleship's API. This is
very early version that doesn't have any configuration or knowledge,
but more here so I have something to work with going forward. I expect
this to change a lot.

edwardloveall committed:
Setup configuration object

This will store anything configurable or needed globally.

edwardloveall committed:
Add base_uri to configuration

In addition to adding the configuration parameter, I also added it to
the Client class for easy access

edwardloveall committed:
Create default base_uri configuration

If no URI is given, which is the default, we use
`https://stattleship.com`

edwardloveall commented on lib/stattleship/client.rb:
```diff
@@ -0,0 +1,45 @@
+    end
+
+    def fetch
+      configure_http
```
Thinking I may put this http stuff in Configuration too.

dthyresson replied:
That makes sense to do.

dthyresson commented on lib/stattleship/version.rb:
```diff
@@ -0,0 +1,5 @@
+module Stattleship
+  module Ruby
+    VERSION = '0.1.0'
```
I've seen cases where people `.freeze` this, but not certain needed.

dthyresson commented on lib/stattleship/configuration.rb:
```diff
@@ -0,0 +1,26 @@
+    attr_writer :base_uri
+
+    def base_uri
+      @base_uri ||= URI('https://stattleship.com')
```
Once we sort out Fastly edge caching, am going to use `api.stattleship.com` but for now I think the base url should be `www.stattleship.com`.

dthyresson commented:
Just a small change to the base_url and good to merge. :shipit:

edwardloveall committed:
Change base uri to www.stattleship.com

edwardloveall committed:
Freeze the version number

It doesn't change at runtime so why not?

edwardloveall committed:
Move http config into configuration.rb

Instead of configuring the http settings inside the client, they have
been moved to the Configuration class

edwardloveall committed:
Refactor tests

Finding and replacing all the stattleship URLs was tedious, so I
decided to DRY them up.
